tensorcircuit/channels.py

```diff
@@ -142,49 +182,30 @@ def generaldepolarizingchannel(
     :return: Sequences of Gates
     :rtype: Sequence[Gate]
     """
-
-    if num_qubits == 1:
-        if isinstance(p, float):
-            probs = [1 - 3 * p] + 3 * [p]
-        elif isinstance(p, Sequence):
-            probs = [1 - sum(p)] + p  # type: ignore
-        else:
-            raise ValueError("p should be float or list")
-    elif num_qubits == 2:
-        if isinstance(p, float):
-            probs = [1 - 15 * p] + 15 * [p]
-        elif isinstance(p, Sequence):
-            probs = [1 - sum(p)] + p  # type: ignore
-        else:
-            raise ValueError("p should be float or list")
+    m = 4**num_qubits - 1
+    if isinstance(p, float):
+        probs = [1 - m * p] + m * [p]
+    elif isinstance(p, Sequence):
+        if not len(p) == m:
+            raise ValueError(f"Invalid probability input {p}")
+        probs = [1 - sum(p)] + list(p)
     else:
-        raise ValueError(f"num_qubits should be 1 or 2. Got {num_qubits}")
+        raise ValueError("p should be float or list")
 
     if not np.all(np.array(probs) >= 0):
         raise ValueError(f"Invalid probability input {p}")
 
-    if num_qubits == 1:
-        tup = [gates.i().tensor, gates.x().tensor, gates.y().tensor, gates.z().tensor]  # type: ignore
-    else:
-        assert num_qubits == 2
-        tup = [
-            gates.ii().tensor,  # type: ignore
-            gates.ix().tensor,  # type: ignore
-            gates.iy().tensor,  # type: ignore
-            gates.iz().tensor,  # type: ignore
-            gates.xi().tensor,  # type: ignore
-            gates.xx().tensor,  # type: ignore
-            gates.xy().tensor,  # type: ignore
-            gates.xz().tensor,  # type: ignore
-            gates.yi().tensor,  # type: ignore
-            gates.yx().tensor,  # type: ignore
-            gates.yy().tensor,  # type: ignore
-            gates.yz().tensor,  # type: ignore
-            gates.zi().tensor,  # type: ignore
-            gates.zx().tensor,  # type: ignore
-            gates.zy().tensor,  # type: ignore
-            gates.zz().tensor,  # type: ignore
-        ]
+    paulis = [gates.i().tensor, gates.x().tensor, gates.y().tensor, gates.z().tensor]  # type: ignore
+    tup = paulis
+    for _ in range(num_qubits - 1):
+        old_tup = tup
+        tup = []
+        for pauli in paulis:
+            for term in old_tup:
+                mat = np.kron(pauli, term).reshape([2, 2] * num_qubits)
```

Comment:
what is the final order, is np.kron(term, pauli) more natural? just a question, not think in detail